elastic/kibana
# [serverless] add cspm metering functionality

solves
- https://github.com/elastic/security-team/issues/7020

First usage of the shared task manager class for the security serverless project:
- Registered CSPM metering callback
- Add logs to the task manager service
- Updated the interfaces 

### How to test it?
1. Follow the instructions provided in: https://github.com/elastic/security-team/issues/7025 to deploy the development serverless environment.
2. Checkout the branch
3. Start Kibana using the following command: `yarn start --serverless=security` (Make sure Elasticsearch is already running before).
4. Check the Kibana logs.
Look for the specified logger: `plugins.securitySolutionServerless.serverless-security:cspm-usage-reporting-task:1`
Review the logs associated with the specified logger to gather the required information.
```bash
[2023-07-20T17:50:38.552+03:00][INFO ][plugins.securitySolutionServerless.serverless-security:cspm-usage-reporting-task:1] usage records report was sent successfully
[2023-07-20T17:51:11.365+03:00][INFO ][plugins.securitySolutionServerless.serverless-security:cspm-usage-reporting-task:1] received usage records: [{"id":"serverless-security:cspm-usage-report","usage_timestamp":"2023-07-20T10:47:54.506Z","creation_timestamp":"7/20/2023, 5:51:11 PM","usage":{"type":"cspm-resource","quantity":1219,"period_seconds":3600,"cause":"CIS Amazon Web Services Foundations"},"source":{"id":"serverless-security:cspm-usage-reporting-task:1","instance_group_id":"missing project id"}}]
[2023-07-20T17:51:11.572+03:00][INFO ][plugins.securitySolutionServerless.serverless-security:cspm-usage-reporting-task:1] usage records report was sent successfully
```
